### PR TITLE
Update to ubuntu 20.04, tcl 8.6 and cpplint 1.6.1 in GitHub Actions

### LINKED ISSUE #941: Github deprecated Ubuntu-18.04 in CI builds.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.

### Motivation

See [GitHub Actions: The Ubuntu 18.04 Actions runner image is being deprecated and will be removed by 12/1/22](https://github.blog/changelog/2022-08-09-github-actions-the-ubuntu-18-04-actions-runner-image-is-being-deprecated-and-will-be-removed-by-12-1-22/)

### Solution

So, maybe we can:

- start to migrate on next version of Ubuntu base images (20.04)
- add Ubuntu 22.04 as second images into matrix

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by aleksraiden**
Preparing CI matrix to use Ubuntu 20.04 Close #941

### CODE REVIEW

**Comment by aleksraiden** [comment]
Lot of thanks @torwig for awesome help and fix cpp linter. Please, review

File: .github/workflows/kvrocks.yaml [hunk 1]
```diff
@@ -79,57 +79,76 @@ jobs:
...
+            with_luajit: -DUSE_LUAJIT=ON
+            with_sanitizer: -DENABLE_ASAN=ON
+            with_openssl: -DENABLE_OPENSSL=ON
+            compiler: gcc
+          - name: Ubuntu GCC TSan with luaJIT
+            os: ubuntu-20.04
+            without_jemalloc: -DDISABLE_JEMALLOC=ON
             with_luajit: -DUSE_LUAJIT=ON
+            with_sanitizer: -DENABLE_TSAN=ON
+            with_openssl: -DENABLE_OPENSSL=ON
```
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml]: I think maybe it is duplicate, could you check these configuration again?

Comment (aleksraiden) [reply .github/workflows/kvrocks.yaml]: Moment...

File: .github/workflows/kvrocks.yaml [hunk 2]
```diff
@@ -79,73 +79,78 @@ jobs:
...
             compiler: clang
-            runtime_env_vars: TSAN_OPTIONS="suppressions=$(realpath ./tests/tsan-suppressions)"
+            runtime_env_vars: TSAN_OPTIONS="suppressions=$(realpath ./tests/tsan-suppressions)"          
           - name: Ubuntu GCC Ninja
-            os: ubuntu-18.04
+            os: ubuntu-20.04
             with_ninja: --ninja
             with_openssl: -DENABLE_OPENSSL=ON
-            compiler: gcc
+            compiler: gcc
```
Comment (tisonkun) [inline .github/workflows/kvrocks.yaml]: Please remove all trailing whitespace.

File: .github/workflows/kvrocks.yaml [hunk 3]
```diff
@@ -79,73 +79,78 @@ jobs:
...
             with_luajit: -DUSE_LUAJIT=ON
             with_sanitizer: -DENABLE_TSAN=ON
             with_openssl: -DENABLE_OPENSSL=ON
             compiler: gcc
-
+          - name: Ubuntu GCC without luaJIT
+            os: ubuntu-20.04
+            without_luajit: -DUSE_LUAJIT=OFF
+            with_openssl: -DENABLE_OPENSSL=ON
+            compiler: gcc
```
Comment (tisonkun) [inline .github/workflows/kvrocks.yaml]: `-DUSE_LUAJIT=OFF` is the default. So this entry can duplicate to `Ubuntu GCC`. Please remove.

**Review by tisonkun — CHANGES_REQUESTED** [review]
Generally looks good. Thank @aleksraiden!

Comments inline.

File: .github/workflows/kvrocks.yaml (hunk 2 above)
Comment (aleksraiden) [reply .github/workflows/kvrocks.yaml]: Thanks, fixed

File: .github/workflows/kvrocks.yaml (hunk 3 above)
Comment (aleksraiden) [reply .github/workflows/kvrocks.yaml]: Done

**Comment by aleksraiden** [comment]
Can't understand, what's are wrong on MacOS builds and tcl tests...

**Comment by tisonkun** [comment]
@aleksraiden Yep.. It's unstable and hard to debug. That's why we try to migrate to go cases #811 . TCL may work fine with experts,  but our community seems no having such ones.

**Comment by aleksraiden** [comment]
Yes @tisonkun your work on Go test are awesome!

**Review by PragmaTwice — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]

**Comment by tisonkun** [comment]
Thank you! Merged.

**Pull request merged**